pkg/search/sorted/searcher.go

```diff
@@ -53,6 +54,52 @@ func Searcher(searcher search.Searcher, field search.FieldLabel, ranker Ranker)
 	}
 }
 
+func IsValidPriorityQuery(q *v1.Query, field search.FieldLabel) (bool, error) {
+	if q.GetPagination() != nil && len(q.GetPagination().GetSortOptions()) == 1 {
+		if q.GetPagination().GetSortOptions()[0].GetField() == field.String() {
+			return true, nil
+		}
+	} else if q.GetPagination() != nil && len(q.GetPagination().GetSortOptions()) > 1 {
+		for _, q := range q.GetPagination().GetSortOptions() {
+			if q.GetField() == field.String() {
+				return false, errors.Errorf("query field %v not supported with other sort options", field.String())
+			}
+		}
+	}
+	return false, nil
+}
+
+func BuildPriorityQuery(q *v1.Query, field search.FieldLabel) (*v1.Query, bool, error) {
```

Comment:
Could this be called RemovePrioritySortFromQuery or something? I think BuildPriorityQuery feels like it just transforms the sort option into something and that user doesn't need to sort the results using ranker afterwards.